src/base/hide_password.cpp

```diff
@@ -0,0 +1,861 @@
+	    {
+	      len = (int) (in - ps);
+	      return ps;
+	    }
+	  break;
+	}
+
+      in++;
+    }
+
+  if (in == ps)
+    {
+      return NULL;
+    }
+
+  len = (int) (in - ps);
+  return ps;
+}
+
+char *
+hide_password::get_method_passowrd_start_position (char *ps, char *method_name, int *method_password_len)
+{
+  /*
+     set_password('password_string')
+     add_user ('user_name', 'password_string')
+     add_user ('user_name')
+     login ('user_name')
+     login ('user_name', 'password')
+   */
+  int len;
+  char *token;
+
+  *method_password_len = -1;
+  token = get_token (ps, len);
+  if (len != 1 || *token != '(')
+    {
+      return NULL;
+    }
+
+  if (strcmp (method_name, "set_password") == 0)
+    {
+      token = get_token (ps, len);	// read user_name
+      if (len >= 2 && (*token == '\'' || *token == '"'))
+	{
+	  *method_password_len = len;
+	  return token;
+	}
+    }
+  else
+    {
+      token = get_token (ps, len);	// read user_name
+      if (len < 2 || (*token != '\'' && *token != '"'))
+	{
+	  return NULL;
+	}
+
+      token = get_token (ps, len);
+      if (len == 1)
+	{
+	  if (*token == ')')
+	    {
+	      *method_password_len = 0;
+	      return token;
+	    }
+	  else if (*token == ',')
+	    {
+	      token = get_token (ps, len);	// read password
+	      if (len >= 2 && (*token == '\'' || *token == '"'))
+		{
+		  *method_password_len = len;
+		  return token;
+		}
+	    }
+	}
+    }
+
+  return NULL;
+}
+
+
+char *
+hide_password::get_passowrd_pos_n_len (char *query, bool is_create, bool is_server, int *password_len,
+				       bool *is_found_pwd_keyword)
+{
+  char *ps = query;
+  char *token, *prev;
+  int len, tlen = 0;
+  bool is_open = false;
+
+  /* pattern cases to consider)
+  * CREATE SERVER srv1 (HOST='localhost', PORT=3300, DBNAME=demodb, USER=dev1, PASSWORD='password_string')
+  * CREATE SERVER srv1 (HOST='localhost', PORT=3300, DBNAME=demodb, USER=dev1, PASSWORD=)
+  * CREATE SERVER srv1 (HOST='localhost', PORT=3300, DBNAME=demodb, USER=dev1, PASSWORD='')
+  * CREATE SERVER srv1 (HOST='localhost', PORT=3300, DBNAME=demodb, PASSWORD=, USER=dev1)
+  * CREATE SERVER srv1 (HOST='localhost', PORT=3300, DBNAME=demodb, USER=dev1)
+  *
+  * CREATE USER user_name
+  * CREATE USER user_name PASSWORD 'password_string'
+  * CREATE USER user_name PASSWORD _utf8'password_string'
+  * CREATE USER user_name PASSWORD _euckr'password_string'
+  * ALTER USER user_name PASSWORD 'password_string'
+  * ALTER SERVER srv1 CHANGE PASSWORD='password string'
+  */
+
+  *password_len = 0;
+  *is_found_pwd_keyword = false;
+  while (*ps)
+    {
+      token = get_token (ps, len);
+      if (*token == ';' || *token == ')')
+	{
+	  return token;
+	}
+
+      if (len == 8 && strncasecmp (token, "password", 8) == 0)
+	{
+	  *is_found_pwd_keyword = true;
+	  break;
+	}
+    }
+
+  if (*ps == '\0')
+    {
+      // check case "CREATE USER user_name"
+      return (is_create && !is_server) ? ps : NULL;
+    }
+
+  prev = NULL;
```

Comment:
trivial:
what about removing this line and initialize it in decl.
char *token, *prev = NULL;

Reply:
I'll keep the current code for this part.
Thank you.